src/shadow-embed.js

```diff
@@ -94,7 +108,8 @@ function createShadowRootPolyfill(hostElement) {
       // Cast to ShadowRoot even though it is an Element
       // TODO(@dvoytenko) Consider to switch to a type union instead.
       /** @type {?}  */ (doc.createElement('i-amphtml-shadow-root')));
-  shadowRoot.id = 'i-amphtml-sd-' + Math.floor(win.Math.random() * 10000);
+  // NOTE (torch2424): Moved shadowRoot.id to createShadowRoot(), as browsers can have a shadowDom version (thefore not calling this fucntion), and not support CSS encapsulation
```

Comment:
Remove this comment and the follow up line. We are all good with moving it to a new place.